lib/ClangImporter/ClangImporter.cpp

```diff
@@ -7471,6 +7471,37 @@ bool importer::isForeignReferenceTypeWithoutImmortalAttrs(const clang::QualType
          !hasImmortalAtts(pointeeType->getDecl());
 }
 
+// TODO: Instead of just checking the immediate parent, do a traversal till the
```

Comment:
I think this should be addressed in this PR. Some things to look out for:
* Diamond inheritance. If we inherit the same refcounted base on different paths that is OK.
* Inheriting multiple refcounted bases. I think that should probably be an error.

Reply:
Actually, I wonder if this is already recursive, since we call the `CxxRecordSemantics` which should also look at the bases. But we should verify that it has the behavior we want for some of the edge cases I mentioned in other review comments.